app/Http/Controllers/OfficerController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\OfficerRequest;
+use App\Models\Officer;
+use Illuminate\Support\Str;
+
+class OfficerController extends Controller
+{
+    public function index()
+    {
+        return view('pages.officers.index',[
+            'officers' => Officer::paginate(20)
+        ]);
+    }
+
+    public function create()
+    {
+        return view('pages.officers.create',[
+            'positions' => Officer::getPositions()
+        ]);
+    }
+
+    public function store(OfficerRequest $request)
+    {
+        if(!Officer::isValidPosition($request->position))
+        {
+            return back()->withErrors(['position' => 'Invalid Position'])->withInput(['fullname' => $request->fullname]);
+        }
+
+        Officer::create([
+            'fullname' => Str::title($request->fullname),
+            'position' => $request->position
+        ]);
+
+        return redirect(route('admin.officers.index'))->with([
+            'created' => true,
+            'message' => 'New Officer has been successfully added!'
+        ]);
+    }
+
+    public function edit(Officer $officer)
+    {
+        $officer = Officer::findOrFail($officer->id);
+
+        return view('pages.officers.edit',[
+            'officer' => $officer,
+            'positions' => Officer::getPositions()
+        ]);
+    }
+
+    public function update(OfficerRequest $request, Officer $officer)
+    {
+        if(!Officer::isValidPosition($request->position))
+        {
+            return back()->withErrors(['position' => 'Invalid Position'])->withInput(['fullname' => $request->fullname]);
+        }
+
+        $officer = Officer::findOrFail($officer->id);
```

Comment:
You don't have to do this as it is already handled by Laravel by binding it in the route. Note above on your parameters list that you already expect that.